### PR TITLE
[transferring] Add mangling support for transferring.

### PR DESCRIPTION
**Pull request by gottesmm**
This includes runtime support for instantiating transferring param/result in function types. This is especially important since that is how we instantiate function types like: typealias Fn = (transferring X) -> ().

rdar://123118061

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: stdlib/public/core/Misc.swift [hunk 1]
```diff
@@ -103,6 +103,7 @@ public func _getMangledTypeName(_ type: Any.Type)
 /// Returns the mangled name for a given type.
 @available(SwiftStdlib 5.3, *)
 @_unavailableInEmbedded
+@_optimize(none)
```
Comment (mikeash) [inline stdlib/public/core/Misc.swift:106]: Guessing this is left over from debugging work.

**Review by mikeash — APPROVED** [review]

File: stdlib/public/core/Misc.swift (hunk 1 above)
Comment (gottesmm) [reply stdlib/public/core/Misc.swift:106]: Thanks!

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test linux platform

**Comment by gottesmm** [comment]
@swift-ci smoke test linux platform

**Comment by gottesmm** [comment]
@swift-ci test linux platform

**Comment by gottesmm** [comment]
@swift-ci smoke test linux platform

**Comment by gottesmm** [comment]
Smoke test failure is unrelated. I am going to see if the full Linux one passes.

**Comment by gottesmm** [comment]
(I think it is already beyond that test, suggesting the test is flaky)

**Pull request merged**